torch/nn/modules/module.py

```diff
@@ -276,6 +293,8 @@ def __call__(self, *input, **kwargs):
                     wrapper = functools.partial(hook, self)
                     functools.update_wrapper(wrapper, hook)
                     grad_fn.register_hook(wrapper)
+        if tracing_state:
+            torch._tracing_state.pop_scope()
```

Comment:
Probably best to try-finally this so that we restore scopes when we pop

Reply:
BTW there is a mistake on my side here (I fixed it on my end but ended up not pushing the fix). It should read
```
if tracing_state:
    tracing_state.pop_scope()
```

Plus try-finally sounds good.